openedx_tagging/core/tagging/api.py

```diff
@@ -96,16 +96,14 @@ def resync_object_tags(object_tags: QuerySet = None) -> int:
 
 
 def get_object_tags(
-    object_id: str, taxonomy: Taxonomy = None, valid_only=True
-) -> Iterator[ObjectTag]:
+    object_id: str, taxonomy_id: str = None
+) -> QuerySet:
     """
-    Generates a list of object tags for a given object.
+    Returns a Queryset of object tags for a given object.
 
     Pass taxonomy to limit the returned object_tags to a specific taxonomy.
-
-    Pass valid_only=False when displaying tags to content authors, so they can see invalid tags too.
-    Invalid tags will (probably) be hidden from learners.
     """
+    taxonomy = get_taxonomy(taxonomy_id)
```

Comment:
@pomegranited I was about to remove it, however the taxonomy object is being used in this line that was introduced related to System Defined Taxonomies:

```py
ObjectTagClass = taxonomy.object_tag_class if taxonomy else ObjectTag
```
Since it was recently added, I figured its still used/needed, or is that no longer the case after the change in scope from product?

Reply:
Uck.. yeah, you are correct @yusuf-musleh : we have to keep that in, at least until it gets refactored out by https://github.com/openedx/modular-learning/issues/85.